File: python/897_递增顺序搜索树_easy.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.dummyHead = TreeNode()
        self.cur = self.dummyHead
    def increasingBST(self, root: TreeNode) -> TreeNode:
        def inorder(node):
            if not node:
                return
            inorder(node.left)
            self.cur.right = node
            node.left = None
            self.cur = node
            inorder(node.right)
        inorder(root)
        return self.dummyHead.right

File: cpp/897_递增顺序搜索树_easy.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    TreeNode *cur;
public:
    void inorder(TreeNode* node) {
        if (node==nullptr) return;
        inorder(node->left);
        cur->right = node;
        node->left = nullptr;
        cur = node;
        inorder(node->right);
    }
    TreeNode* increasingBST(TreeNode* root) {
        TreeNode* dummyHead = new TreeNode();
        cur = dummyHead;
        inorder(root);
        return dummyHead->right;
    }
};